### PR TITLE
Imp: replace go/net/url

### PR DESCRIPTION
**Pull request by christophe-liwei**
根据[#769](https://github.com/apache/dubbo-go/pull/769)的讨论，替换`go/net/url`为`common/net/url`。

```
// Set sets the key to value.
// If value is not "", it will replaces existing values.
func (v Values) Set(key, value string) {
	if len(value) == 0 {
		return
	}
	v[key] = []string{value}
}

// Add adds the value to key. If value is not "", it appends to existing
// values associated with key.
func (v Values) Add(key, value string) {
	if len(value) == 0 {
		return
	}
	v[key] = append(v[key], value)
}
```

Values的Set方法替换为value 是非空字符串时才能替换old value

### CODE REVIEW

File: common/url.go [hunk 1]
```diff
@@ -20,9 +20,9 @@ package common
 import (
 	"encoding/base64"
 	"fmt"
+	"github.com/apache/dubbo-go/common/net/url"
```
Comment (AlexStocks) [inline common/url.go:23]: move this package to the third import block

**Review by AlexStocks — APPROVED** [review]

**Review by zouyx — COMMENTED** [review]
Target branch to develop 
Thanks

**Comment by IndiraFinish** [comment]
ok @zouyx

File: common/url_test.go [hunk 2]
```diff
@@ -41,6 +41,9 @@ func TestNewURLWithOptions(t *testing.T) {
 	methods := []string{"Methodone,methodtwo"}
 	params := url.Values{}
 	params.Set("key", "value")
+	// test value is ""
+	params.Set("group", "")
+	params.Set("vserion", "")
```
Comment (pantianying) [inline common/url_test.go:46]: vserion

File: common/net/url/url.go [hunk 3]
```diff
@@ -0,0 +1,1161 @@
...
+//
+// Note that the Path field is stored in decoded form: /%47%6f%2f becomes /Go/.
+// A consequence is that it is impossible to tell which slashes in the Path were
+// slashes in the raw URL and which were %2f. This distinction is rarely important,
+// but when it is, the code should use RawPath, an optional field which only gets
+// set if the default encoding is different from Path.
+//
+// URL's String method uses the EscapedPath method to obtain the path. See the
+// EscapedPath method for more details.
+type URL struct {
```
Comment (zouyx) [inline common/net/url/url.go:367]: Can not extends net/URL？

**Review by zouyx — COMMENTED** [review]
Fix conflict and change title based on contribution.md

**Comment by watermelo** [comment]
hi @IndiraFinish, pls resolve the conflicts.

**Comment by IndiraFinish** [comment]
I'm sorry @watermelo,I didn't bring my computer today

File: common/net/url/url.go (hunk 3 above)
Comment (IndiraFinish) [reply common/net/url/url.go:367]: I agree with you that it was thoughtless here

Comment (hxmhlt) [reply common/net/url/url.go:367]: +1

**Review by zouyx — COMMENTED** [review]
Merge develop and make Checks to normal.

**Comment by zouyx** [comment]
what about this feature？

**Comment by AlexStocks** [comment]
> what about this feature？

@haohongfan own it now.

**Pull request closed without merging**